### PR TITLE
feat(graph-db): add v4 graph-db service with sqlite ensure

### PR DESCRIPTION
**Pull request by subtleGradient**
## Summary
- add new @effect-native/graph-db package scaffold and public API
- implement makeGraphDb(spec) service factory, node/edge repos, and schema-plan types/errors
- implement SQLite dialect with expand-only ensure planning/execution and CRR plan constraints
- add v4 guardrail docs, README usage, and example
- add package test suite for ensure, transactions, node roundtrip, edge traversal, and CRR plan assertions

## Verification
- cd packages/graph-db && bun run check
- cd packages/graph-db && bun test
- bun run lint-fix

### CODE REVIEW

File: packages/graph-db/src/repos/edgeRepo.ts [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+  readonly spec: GraphSpec
+  readonly sql: SqlClient.SqlClient
+}
+
+export const makeEdgeRepository = ({ spec, sql }: EdgeRepositoryOptions): EdgeRepository => {
+  const tableName = spec.edgeTableName ?? "graph_edge"
+
+  const put: EdgeRepository["put"] = (edgeType, src, dst, props) =>
+    Effect.gen(function*() {
+      const encodedProps = props === undefined ? null : JSON.stringify(props)
```
[the remark this answers is not in the document]
Comment (subtleGradient) [reply packages/graph-db/src/repos/edgeRepo.ts]: Fixed in 01283596f. `edgeRepo.put` now encodes props through `encodePropsJson` using `Effect.try`, so `JSON.stringify` failures (circular structures, `BigInt`, etc.) fail through the typed error channel as `GraphInvariantError` (`context: "edge.encode"`) instead of defecting. Added a regression test in `packages/graph-db/test/edge-repo.traversal.test.ts` (`returns a typed failure when edge props are not JSON-serializable`). Verified with `bun test packages/graph-db/test`.

File: packages/graph-db/src/ensure/GraphDialectSqlite.ts [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+        reason: "UniqueIndexForbiddenInCrr",
+        detail: `Secondary unique index ${index.name} is not allowed in CRR mode`
+      })
+      continue
+    }
+
+    const existing = existingIndexes.get(index.name)
+    if (!existing) {
+      createIndexStatements.push(createIndexSql(table.name, index))
+      continue
```
[the remark this answers is not in the document]
Comment (subtleGradient) [reply packages/graph-db/src/ensure/GraphDialectSqlite.ts:356]: Fixed in 01283596f. `GraphDialectSqlite` now reads schema-wide index ownership from `sqlite_master` and marks cross-table same-name reuse as an incompatible plan (`IndexNameCollision`) before emitting any `CREATE INDEX` statement. Added a regression test in `packages/graph-db/test/sqlite-dialect.ensure.test.ts` (`fails when an index name is already used by a different table in the same schema`). Verified with `bun test packages/graph-db/test`.

File: packages/graph-db/src/repos/edgeRepo.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (subtleGradient) [reply packages/graph-db/src/repos/edgeRepo.ts]: Re-verified on current HEAD (`c3082a2b1`): still resolved. `edgeRepo.put` encodes props through `encodePropsJson` with `Effect.try`, so non-serializable inputs fail with typed `GraphInvariantError` (`context: "edge.encode"`) instead of defecting. Regression remains green: `bun test packages/graph-db/test` includes `returns a typed failure when edge props are not JSON-serializable`.

File: packages/graph-db/src/ensure/GraphDialectSqlite.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (subtleGradient) [reply packages/graph-db/src/ensure/GraphDialectSqlite.ts:356]: Re-verified on current HEAD (`c3082a2b1`): still resolved. `GraphDialectSqlite` detects schema-global index name collisions via `sqlite_master` ownership checks and returns `IndexNameCollision` incompatibilities before create. Regression remains green: `bun test packages/graph-db/test` includes `fails when an index name is already used by a different table in the same schema`.
